sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs

```diff
@@ -15,13 +15,37 @@ namespace Azure.AI.OpenAI
     /// <summary> Azure OpenAI APIs for completions and search. </summary>
     public partial class OpenAIClient
     {
+        private const string publicOpenAIVersion = "1";
+        private const string publicOpenAIEndpoint = $"https://api.openai.com/v{publicOpenAIVersion}";
+
+        private TokenCredential _tokenCredential;
+        private HttpPipeline _pipeline;
+        private Uri _endpoint;
+
+        private string _publicOpenAIToken = string.Empty;
+
+        public string PublicOpenAIToken
```

Comment:
I don't think this needs to be public does it?

Reply:
If I'm reading correctly, it does need to be public because it's how the customer using the API would configure the client into "non-AOAI public endpoint mode." It could be an init-only setter if we desired (C# version permitting) but this is effectively the light switch.

Reply:
Travis is correct, this is the light switch for setting Client to use the public OpenAI endpoint instead of Azure.